### PR TITLE
BI-3925 Add authentication and session middleware

### PR DESCRIPTION
**Pull request by markpit**
Adding CH session and authentication middleware

### CODE REVIEW

File: src/middleware/authentication.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { authMiddleware } from "web-security-node";
+import * as pageURLs from "../model/page.urls";
+
+const authMiddlewareConfig = {
+  accountWebUrl: "",
+  returnUrl: `${pageURLs.STRIKE_OFF_OBJECTIONS}${pageURLs.COMPANY_NUMBER}`,
+};
+
+const signInMiddleware = authMiddleware(authMiddlewareConfig);
```
Comment (mwestacott) [inline src/middleware/authentication.middleware.ts:9]: Perhaps call this authenticationMiddleware and export/import it as this. Then consistent with sessionMiddleware, below.

File: src/app.ts [hunk 2]
```diff
@@ -1,9 +1,13 @@
+import cookieParser from "cookie-parser";
 import express from "express";
 import * as nunjucks from "nunjucks";
 import * as path from "path";
 import { checkServiceAvailability } from "./availability/middleware/service.availability";
 import { APP_NAME } from "./constants";
 import logger from "./logger";
+import signInMiddleware from "./middleware/authentication.middleware";
```
Comment (mwestacott) [inline src/app.ts:8]: See suggested import name change below.

File: src/middleware/authentication.middleware.ts (hunk 1 above)
Comment (markpit) [reply src/middleware/authentication.middleware.ts:9]: will do

File: src/app.ts (hunk 2 above)
Comment (markpit) [reply src/app.ts]: done

File: src/middleware/authentication.middleware.ts (hunk 1 above)
Comment (markpit) [reply src/middleware/authentication.middleware.ts]: done

**Pull request merged**